sycl/source/handler.cpp

```diff
@@ -14,23 +14,103 @@
 #include <CL/sycl/event.hpp>
 #include <CL/sycl/handler.hpp>
 #include <CL/sycl/info/info_desc.hpp>
+#include <detail/kernel_bundle_impl.hpp>
 #include <detail/kernel_impl.hpp>
 #include <detail/queue_impl.hpp>
 #include <detail/scheduler/scheduler.hpp>
 
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
+context handler::getContext() { return MQueue->get_context(); }
+
+std::shared_ptr<detail::kernel_bundle_impl>
+handler::getOrInsertHandlerKernelBundle(bool Insert) {
```

Comment:
Probably not worth it, but wouldn't it be better to have different functions for get and insert?

Reply:
In this case insert function would duplicate logic of get function because the whole operation should be done under mutex.